main/accounts/index.ts

```diff
@@ -521,16 +522,28 @@ export class Accounts extends EventEmitter {
     const request = currentAccount.requests[handlerId] && currentAccount.requests[handlerId].type === 'transaction'
     if (!request) return cb(new Error(`Could not locate request ${handlerId}`))
 
+    const lastSignerType = getSignerType(currentAccount.lastSignerType)
     const signer = currentAccount.getSigner()
-    if (!signer) return cb(new Error('No signer'))
-
-    if (signer.status === 'locked') {
-      const crumb = {
-        view: 'expandedSigner', 
-        data: { signer: signer.id }
+    const isHardware = isHardwareSigner(lastSignerType)
+    
+    // handle locked and other states requiring user action
+    if (!signer && isHardware || signer && !isHardware) {
```

Comment:
Yes

Reply:
no it doesn't, I just tested with my locked Ledger Nano S and it just sits there as before. I'm reading through the logic and it makes sense why it wouldn't work. if the signer is hardware it will only ever go into this block if there's no signer

Reply:
Interesting, I'm pretty sure I tested this before

Reply:
Just tested with locked Nano S and WFM.